bin/pkg.ml

```diff
@@ -228,12 +271,13 @@ module Lock = struct
        effects after performing validation so that if materializing any
        lockdir would fail then no side effect takes place. *)
     let write_disk_list =
-      List.map per_context
-        ~f:(fun { Per_context.lock_dir_path; version_preference } ->
+      List.mapi per_context
+        ~f:(fun i { Per_context.lock_dir_path; version_preference; _ } ->
           let summary, lock_dir =
             Dune_pkg.Opam.solve_lock_dir ~version_preference ~repo_selection
               opam_file_map
           in
+          if i > 0 then Console.printf "";
```

Comment:
I just noticed this, but isn't it weird that print these messages before actually writing anything to disk?

Reply:
In any case, you can improve `summary` to return ` _ Pp.t` and then use `Pp.concat` with the appropriate `~sep` to add the spacing you need.

Reply:
Ok I've updated this to defer printing until after this loop and adding newlines with `Pp.concat`.